src/h5web/visualizations/heatmap/utils.ts

```diff
@@ -151,3 +60,14 @@ export function generateCSSLinearGradient(
 
   return `linear-gradient(to ${direction}, ${gradientColors})`;
 }
+
+export function getColorScale(
+  domain: [number, number] | undefined,
+  logScale: boolean
+): ColorScale | undefined {
+  if (domain === undefined) {
+    return undefined;
+  }
+
+  return logScale ? scaleSymlog().domain(domain) : scaleLinear().domain(domain);
```

Comment:
The domain is set to the bounds of the data but the range needs not to be set as:
- in `computeTextureData`: the range is irrevelevant as it is mapped directly to the domain of the `scaleSequential`.
- in the axis of `ColorBarAxis`; the range is set to the height of the gradient.

Reply:
You can simplify this: 

```ts
const scaleFunction = logScale ? scaleSymlog : scaleLinear;
return scaleFunction().domain(domain);
```

Reply:
I had to add `as ColorScale` to the return value to make TS happy. There are fishy things with d3-scale types...